hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8303,9 +8428,14 @@ public void processRowsWithLocks(RowProcessor<?,?> processor, long timeout,
             prevRowLock = rowLock;
           }
         }
+
         // STEP 3. Region lock
         lock(this.updatesLock.readLock(), acquiredRowLocks.isEmpty() ? 1 : acquiredRowLocks.size());
         locked = true;
+
+        // From this point until memstore update this operation should not be interrupted.
+        disableInterrupts();
```

Comment:
Same with the doMiniBatchMutation above.

Reply:
Will add a checkInterrupt here too.

Reply:
This code is not executed in a loop so removing the thread from the map would be fine. However the reason why I also do disable/enable interrupt is as follows: We already have to do this for doMiniBatchMutation. The disable/enable interrupt pair is a new code discipline. This section of the code also has the same requirements. Apply the new code discipline here too for consistency.